catalog/rest/rest.go

```diff
@@ -82,7 +82,7 @@ var (
 
 func init() {
 	reg := catalog.RegistrarFunc(func(name string, p iceberg.Properties) (catalog.Catalog, error) {
-		return newCatalogFromProps(name, p.Get("uri", ""), p)
+		return newCatalogFromProps(context.Background(), name, p.Get("uri", ""), p)
 	})
```

Comment:
if we're going to use a context in all of these, then yea we should definitely change the definition of the registry and `LoadCatalog` to accept a context

Reply:
👍